### PR TITLE
GaAs example

### PR DESCRIPTION
**Pull request by mfherbst**
Adds as example constructing a GaAs surface using ase.

### CODE REVIEW

File: examples/GaAs.jl [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+atoms = load_atoms(surface)
+atoms = [ElementPsp(el.symbol, psp=load_psp(el.symbol, functional="pbe")) => position
+         for (el, position) in atoms]
+
+lattice = load_lattice(surface)
+model = model_DFT(lattice, atoms, [:gga_x_pbe, :gga_c_pbe])
+basis = PlaneWaveBasis(model, Ecut; kgrid=kgrid)
+
+scfres = self_consistent_field(basis, tol=1e-8,
+                               n_bands=ceil(Int, model.n_electrons / 2),
```
Comment (antoine-levitt) [inline examples/GaAs.jl:37]: I usually do `div(N,2)`

Comment (antoine-levitt) [reply examples/GaAs.jl:37]: also maybe not necessary?

**Comment by antoine-levitt** [comment]
Great! Rename to GaAs_surface maybe?

**Comment by antoine-levitt** [comment]
Also mention the paper on TF mixing for the source

File: examples/GaAs.jl (hunk 1 above)
Comment (mfherbst) [reply examples/GaAs.jl:37]: no this is leftover from my playing. I'll remove it.

**Comment by antoine-levitt** [comment]
Can you also add that this needs `ase`? Or maybe we put it somewhere in the README that ASE is useful?

**Comment by antoine-levitt** [comment]
Reading the code also it's not immediately clear that `load_atoms` refers to ASE. Since this is python it's not clear we emulate dispatch here, so probably it's better to have explicit `load_atoms`? This is just nitpicking though.

**Comment by mfherbst** [comment]
No I disagree. This actually saves code because abipy structures are also pymatgen structures so dispatch works for them, too. The same is true for various kinds of Atoms you can make out of ASE atoms etc. Also `load_atoms` is just "the" function to convert to DFTK atoms no matter where you are coming from, so it is not an unexpected interface at all.

**Comment by antoine-levitt** [comment]
Alright

**Pull request merged**